tests/virt_control_plane_test.go

```diff
@@ -164,27 +165,25 @@ var _ = Describe("[Serial][ref_id:2717][sig-compute]KubeVirt control plane resil
 			eventuallyWithTimeout(waitForDeploymentsToStabilize)
 		})
 
-		When("evicting pods of control plane", func() {
-			test := func(podName string) {
-				By(fmt.Sprintf("Try to evict all pods %s\n", podName))
-				podList, err := getPodList()
-				Expect(err).ToNot(HaveOccurred())
-				runningPods := getRunningReadyPods(podList, []string{podName})
-				Expect(len(runningPods)).ToNot(Equal(0))
-				for index, pod := range runningPods {
-					err = virtCli.CoreV1().Pods(flags.KubeVirtInstallNamespace).Evict(context.Background(), &v1beta1.Eviction{ObjectMeta: metav1.ObjectMeta{Name: pod.Name}})
-					// trying to evict the last running pod in this list should fail
-					if index == len(runningPods)-1 {
-						Expect(err).To(HaveOccurred(), "no error occurred on evict of last pod")
-					} else {
-						Expect(err).ToNot(HaveOccurred())
-					}
+		table.DescribeTable("when evicting pods of control plane", func(podName string) {
+			By(fmt.Sprintf("Try to evict all pods %s\n", podName))
+			podList, err := getPodList()
+			Expect(err).ToNot(HaveOccurred())
+			runningPods := getRunningReadyPods(podList, []string{podName})
+			Expect(len(runningPods)).ToNot(Equal(0))
+			for index, pod := range runningPods {
+				err = virtCli.CoreV1().Pods(flags.KubeVirtInstallNamespace).Evict(context.Background(), &v1beta1.Eviction{ObjectMeta: metav1.ObjectMeta{Name: pod.Name}})
+				// trying to evict the last running pod in this list should fail
+				if index == len(runningPods)-1 {
+					Expect(err).To(HaveOccurred(), "no error occurred on evict of last pod")
```

Comment:
I know you are not making any change here, but can you share which error is expected, and why do we want this error so much that we test for it?
If we evict our first running pod and k8s is very fast to reschedule it, would we still see an error on eviction of our last pod?

Reply:
PodDisruptionBudget is expected to protect the last pod. I will take a look if `client-go`  does expose typed error so we can assert better here(or assert on message).

Yes, your assumption is correct but it is very unlikely to happen because the Pod is expected to be ready not only created afaik.

Reply:
Improved.

Reply:
Thanks. Now it's clearer to me what the test is aimed to be testing. But the down side is that it is apparent that we test k8s, not k6t. Would it be correct to say that the true functionality that we aim for is that IF all virt-api and virt-controller pods are evicted, virt-api is still functional?

I may be wrong, but I feel that a better functional test would test in that level: start a VM, stop it, evict all pods (maybe via a label?), see that the VM can start just as quickly as before. (just an idea/question, not in this PR)

Reply:
>Thanks. Now it's clearer to me what the test is aimed to be testing. But the down side is that it is apparent that we test k8s, not k6t. 

I don't agree we are testing k8s instead of k6t. We are testing integration with k8s. We know what is needed in order to have at least one `virt-api`avaible for running **now**.

It is true that we could just test that we created pdb but it is also beneficial to test the whole flow/integration here because we could set the wrong labels, k8s could change or have bugs, and so on.

>Would it be correct to say that the true functionality that we aim for is that IF all virt-api and virt-controller pods are evicted, virt-api is still functional?

We are testing that `virt-api` is functional because we protect ourself from evicting all `virt-api` pods.